### DISCUSSION TITLE
Add Remotion as a supported framework

### DISCUSSION THREAD

**Discussion by JonnyBurger** [vercel/community#6175] [Idea]
I propose to add [Remotion](http://remotion.dev/) as a [supported framework](https://vercel.com/docs/frameworks/more-frameworks) on Vercel, meaning it can be deployed using zero configuration.

## About Remotion
It's a framework for designing videos programmatically in React. It has around 200.000 downloads per month on NPM.

If you have a server, you can also render the videos into an MP4 file (or many other formats). Since recently, we support deploying our Preview interface "Remotion Studio" to Vercel. However at the moment, one needs to manually enter build commands.

## Why it's cool
- The only React framework for creating videos
- You can continuously deploy changes to your video and get preview links for your changes
- Use the web interface to change the props of the video ([demo](https://watercolor-map.vercel.app))
- You can render videos in the terminal, Node.JS or Lambda by passing the Vercel deployment URL! ([example 1](https://twitter.com/JNYBGR/status/1752311271825789068), [example 2](https://twitter.com/JNYBGR/status/1752332483838775754)).
- We recognize `VERCEL_GIT_REPO_SLUG` and use it to [link back to the source code of components](https://www.remotion.dev/blog/deployable-studio#explore-code-visually)

## Future plans
Remotion is stable as a company, has enough capacity to maintain the framework forward and we have many ideas for future enhancements.

- Making the whole UI responsive
- Generating CLI render commands and Node.JS snippets directly from the UI
- Potentially integrating with the Vercel Build Output API to trigger renders on Lambda and Cloud Run
- (Long-term) offer a browser rendering solution where a video can be rendered using only the frontend
- (Long-term) Render directly on Vercel Serverless Functions (if in the future Vercel adds support for Lambda Layers)

## Pull request

https://github.com/vercel/vercel/pull/11298

**Comment by clearlysid** [comment]
Love using Remotion and excited to see this come to life eventually 🥳

**Comment by shrix1** [comment]
Excited about this

**Comment by thecmdrunner** [comment]
+1 for this. I have some cool stuff to try myself 🤩

**Comment by alantriesagain** [comment]
Remotion is too good for vercel, I'm actually glad they ignored.
